crates/hir_ty/Cargo.toml

```
@@ -18,9 +18,9 @@ ena = "0.14.0"
 log = "0.4.8"
 rustc-hash = "1.1.0"
 scoped-tls = "1"
-chalk-solve = { version = "0.64", default-features = false }
-chalk-ir = "0.64"
-chalk-recursive = "0.64"
+chalk-solve = { version = "0.67", default-features = false }
+chalk-ir = "0.67"
+chalk-recursive = "0.67"
```

```suggestion
chalk-recursive = { version = "0.67", default-features = false }
```

I think this version of `chalk-recursive` has https://github.com/rust-lang/chalk/pull/703 included in it, so the extra deps mentioned in https://github.com/rust-lang/chalk/issues/686 can now be avoided

---

I think people sometimes use those to debug `chalk`, like in https://gist.github.com/detrumi/179079d7065bb4f5720797b1cdbed646.

CC @jackh726, @detrumi, @matklad

---

I think we don't setup tracing in our `main`, so these cant' help for the actually published rust-analyzer binary. If you need to re-build from source, I think it's fine if you change cargo toms to enable logging

---

We enable tracing if CHALK_DEBUG is set.

---

Coming  back to this, I belive memoryruins's suggestion is correct -- chalk-recursive uses `tracing` unconditionally, it's only the subscriber part that's opt-in, and we enable that on our side. submitted #9273